### PR TITLE
Refactor Number Visual

### PR DESCRIPTION
**Pull request by karimkawambwa**
## Description

Refactor number visual:

```
      <NumberChart
        title={text('Title', 'Income')}
        data={data}
        labels={d => format(d.y)}
      />
```

## Type of change

- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Screenshots

Screen Shot 2020-03-17 at 10 50 22 AM

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

**Comment by karimkawambwa** [comment]
@kilemensi 

> Not sure about the NumberChart name... I think Number, Stat, Fact, Figure, Highlight, Spotlight, etc. are all better names. e.g. HURUmap UI is a library for charts, snippets and figures.

AM okay with any name, i though NumberChart sounded fine. But let me know which one you prefer more and we will choose from there. @KhadijaMahanga and @nerdlet  which one do you guys prefer as-well ?

> 2\. So I think we should name the value/aside/note props to reflect this: May be _value_ should be `label` and _aside_/_note_ should be `value(s)`.

I find `values` to be very un targeted and confusing.   Maybe `hover: { aside }`.

**Comment by kilemensi** [comment]
> I find values to be very un targeted and confusing. Maybe hover: { aside }.

That's how it should be @karimkawambwa ... How something is displayed should be up to the component and not prop names e.g. the same `data` prop works for Line, Bar and Pie charts but each ends up displaying it differently. This is also how I see the `value` and `aside/note` props.

If `values` is still too confusing, we can go back to `data` i.e. this `label` is backed with this `data`.

File: packages/charts/src/NumberChart.js [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+          )}
+        </Typography>
+        {typeof hover !== 'string' && hover && hover.bottom && (
+          <Typography className={classes.note}>{hover.bottom}</Typography>
+        )}
+      </div>
+
+      {label && (
+        <Typography className={classes.description}>{label}</Typography>
+      )}
```
Comment (kilemensi) [inline packages/charts/src/NumberChart.js]: Looks like we allow showing 2 labels: One from data and one from labels function.
Screenshot from 2020-03-17 17-05-05

What's the logic behind it?

**Comment by karimkawambwa** [comment]
> * Thinking about this, I believe the idea of this component was just giving it data and ask it to display the data in a certain way. For example if I give it data and ask it to display the data as %, _then_ the real numbers will be displayed on hover, otherwise no hover state if the displayed label matches data. Similar to when I display a non-numeric label like language name, hover will be shown to show the data behind that label... So it's not like you set a manual hover, etc.,
> * I can't remember why we need two hover states (right, and bottom) (or could it be that the hover should be displayed below the label if it doesn't fit on the right?)

@kilemensi  am not sure either, perhaps the other hover was because of a newline and it was assumed to be a secondary hover. @nerdlet maybe you can let us know.

PS:

Giving all this, could someone take lead and produce the requirements for this component. @kilemensi

**Comment by kilemensi** [comment]
> > * Thinking about this, I believe the idea of this component was just giving it data and ask it to display the data in a certain way. For example if I give it data and ask it to display the data as %, _then_ the real numbers will be displayed on hover, otherwise no hover state if the displayed label matches data. Similar to when I display a non-numeric label like language name, hover will be shown to show the data behind that label... So it's not like you set a manual hover, etc.,
> > * I can't remember why we need two hover states (right, and bottom) (or could it be that the hover should be displayed below the label if it doesn't fit on the right?)
> 
> @kilemensi am not sure either, perhaps the other hover was because of a newline and it was assumed to be a secondary hover. @nerdlet maybe you can let us know.
> 
> PS:
> 
> Giving all this, could someone take lead and produce the requirements for this component. @kilemensi

Yes, I nominate @karimkawambwa :sunglasses:

**Comment by nerdlet** [comment]
@karimkawambwa @kilemensi That was the design expected. Some of the hover elements were supposed to go to the next line.

**Comment by kilemensi** [comment]
Migrating to [Wazimap NG](https://github.com/OpenUpSa/wazimap-ng) which means we'll need to re-architect the whole of HURUmap UI around its [UI](https://github.com/OpenUpSa/wazimap-ng-ui)

**Comment by karimkawambwa** [comment]
Wazimap NG looks very slick

**Pull request closed without merging**